pkg/apis/configuration/validation/validation.go

```diff
@@ -801,6 +801,10 @@ func validateRedirectURL(redirectURL string, fieldPath *field.Path, validVars ma
 		return append(allErrs, field.Required(fieldPath, "must specify a url"))
 	}
 
+	if !strings.Contains(redirectURL, "://") {
+		return append(allErrs, field.Required(fieldPath, "must contain the protocol with '://', for example http://, https:// or ${scheme}://"))
```

Comment:
For `field.Required`: 
```go
// Required returns a *Error indicating "value required".  This is used
// to report required values that are not provided (e.g. empty strings, null
// values, or empty arrays).
```
My interpretation of the above would be that `field.Required` should only be used if `url` was unset, empty string, null, etc. 
Considering `url` is set, but an incorrect format I would think that `field.Invalid` is more appropriate.

_Hm I noticed a similar use of `field.Required`. Perhaps I'm mistaken:_
https://github.com/nginxinc/kubernetes-ingress/blob/c2bccae7a052538d2c7fc96a6f891ad45e2d9101/pkg/apis/configuration/validation/validation.go#L1329

What do you think @Rulox ? Happy to be corrected.

Reply:
Yes it should be Invalid